### DISCUSSION TITLE
Inline git diff by buffer

### DISCUSSION THREAD

**Discussion by wh75er** [helix-editor/helix#9235] [Ideas]
#405 issue suggests to add a side by side file diffing (with git diffs in mind as I understood)

I personally prefer inline git/file diffing. It is more convenient to use, e.g. go to definition feature and easier to read (IMO)

<img width="608" alt="image" src="https://github.com/helix-editor/helix/assets/26790838/7981afc2-7197-4fb4-a618-601c8b9de629">

It would be great to have a feature that allows to see a diff with master or any other user specified git branch

limitations (I guess):
- deleted lines:
  - would be shown as a virtual lines (next, as a consequence)
  - would not have working go to definition function or any code actions
  - syntax highlighting? (would it be possible to apply tree-sitter on those virtual lines?)

**Comment by pascalkuthe** [comment]
Lets not keep adding issues for slight variants of features we already have issues for/are already working on. An idea like this would be suited to a comment on an issue and does not need a separate issue. That way we just endup with a million slightly different feature requests.

We are also unlikely to implement 5 different diff modes so the decision which design to use will be made during implementation (and informed by testing). So when we close #405 we will either have implemented this or decided against this approach so tracking separately makes no sense.

**Reply by MadelineBaggins** [reply]
With helix's new inline displays for lsp diagnostics. This could also be achieved by implementing a language server, perhaps with a code action to toggle its effects on or off. Perhaps not relevant entirely here but I just had that thought and it seemed interesting enough to share :)
